src/urllib3/connectionpool.py

```diff
@@ -246,7 +247,7 @@ def _get_conn(self, timeout=None):
             if self.block:
                 raise EmptyPoolError(
                     self,
-                    "Pool reached maximum size and no more connections are allowed.",
+                    "No connection is available, and new connection cannot be created due to blocking mode.",
```

Comment:
I was very confused by the original error message. How can I be getting EmptyPoolError if the pool has reached maximum size? In reality, what I would tell the user is "hey, you are in blocking mode and you have already checked out all maxsize conns so we cannot create a new conn for you." This is what I changed the message to, except, you know, more formal-sounding.